### DISCUSSION TITLE
ESP32 BLE Advertiser Example - iOS (iPhone/Mac)

### DISCUSSION THREAD

**Discussion by tribalmatt** [Moddable-OpenSource/moddable#778] [Q&A] [answered]
Hi team,

I am finally back writing code and exploring and I need to connect a phone to an esp32 over some wireless method.  

I went and built the example "ble/advertiser" and ran this on a esp32 device.  I then opened the settings on my iPhone and looked for a device being published there.  I did not see anything?  I also checked the BlueSee app and did not see anything here.  There were some notes from 2018 that Ben sent out about config files.  But I think this are dated now.  I do load in xsBug properly with no errors.

Is there something obvious I am missing?  Are my expectations wrong?  I don't have my Moddable 2 with me or the book to check these resources.  

Thanks for any pointers,
Matt

**Comment by phoddie** [comment]
> I am finally back writing code...

A good way to start the year.

> I also checked the BlueSee app

I'm not familiar with BlueSee. Perhaps try using nRF Connect or LightBlue. Both see the example advertisement as expected.

<img src=https://user-images.githubusercontent.com/6856458/149845789-6eead443-829d-41a9-8bae-508b8eac7b7b.png width=300/>

If that doesn't work, you might also try seting breakpoints in xsbug just to be sure that `startAdvertising` is called as expected.

**Comment by tribalmatt** [comment]
Thanks for the note back.  I did a clean rebuild of everything and
reinstalled the esp-idf sdk and now everything is working.  I am not sure
what I had wrong, but it works as expected now.

Lastly I tried updating to the 4.3.2 esp-idf and that currently seems to
have some configuration mismatches with moddable sdk.  Is that correct?  I
did not see any notes about this in the discussion.  This was on Mac OS X

Thx,
Matt

**Comment by phoddie** [comment]
Glad to hear BLE is working for you.

ESP-IDF 4.3.2 should work without problems. That is what our documentation [recommends](https://github.com/Moddable-OpenSource/moddable/blob/public/documentation/devices/esp32.md#installing) and what I'm currently using. Sometimes updating the ESP-IDF doesn't succeed as there are lots of sub-repositories. You might just try a clean pull of the full ESP-IDF:

```
cd ~/esp32
git clone -b v4.3.2 --recursive https://github.com/espressif/esp-idf.git
```

**Comment by tribalmatt** [comment, marked as answer]
Thanks again. I forgot I was on my PR branch from back in October.  So my
documentation said to use 4.3.1.  Which of course made sense based on that
date.  I rebased and all is good.
